demos/text_detection_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,408 @@
+}
+
+GAPI_OCV_KERNEL(OCVCropLabels, custom::CropLabels) {
+    static void run(const cv::Mat& image,
+                    const std::vector<cv::RotatedRect>& rrs,
+                    const std::vector<size_t>& outDims,
+                    const bool centralCrop,
+                          std::vector<cv::Mat>& out,
+                          std::vector<std::vector<cv::Point2f>>& pts) {
+        cv::Size outSize(outDims[3], outDims[2]);
+        std::vector<int> blobShape(4);
+        std::copy(outDims.begin(), outDims.end(), blobShape.begin());
+        const std::vector<cv::Point2f> imgPts { { 0.0f,              0.0f              },
+                                                { image.cols - 1.0f, 0.0f              },
+                                                { image.cols - 1.0f, image.rows - 1.0f },
+                                                { 0.0f,              image.rows - 1.0f } };
+        out.clear();
+        out.reserve(rrs.size());
+        pts.clear();
+        pts.reserve(rrs.size());
+        cv::Mat crop(outSize,   CV_8UC3,  cv::Scalar(0));
+        cv::Mat blob(blobShape, CV_32FC1, cv::Scalar(0));
+        std::vector<cv::Point2f> points(4);
+        for (auto& rr : rrs) {
+            if (rr.size != cv::Size2f(0, 0)) {
+                rr.points(points.data());
+                cropImage(image, points, outSize, crop);
+            } else {
+                if (centralCrop) {
+                    auto r = centralCropRect(image.cols, image.rows);
+                    points = centralCropPoints(r);
+                    cv::resize(image(r), crop, outSize);
+                } else {
+                    points = imgPts;
+                    cv::resize(image, crop, outSize);
+                }
+            }
+            crop.reshape(1, blobShape).convertTo(blob, CV_32F);
+            out.emplace_back(blob.clone());
+            pts.push_back(points);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL_ST(OCVCompositeTRDecode, custom::CompositeTRDecode, cv::GComputation) {
+    static void setup(const cv::GArrayDesc&, const cv::GArrayDesc&, const cv::gapi::GNetPackage&,
+                      const size_t, const size_t, std::shared_ptr<cv::GComputation>& graph) {
+        cv::GMat inDec, inHidden, feature, outHidden, outDec;
+        std::tie(outHidden, outDec) =
+            cv::gapi::infer<nets::TextRecognitionDecoding>(inDec, inHidden, feature);
+        graph = std::make_shared<cv::GComputation>(cv::GIn(inDec, inHidden, feature),
+                                                   cv::GOut(outHidden, outDec));
+    }
+    static void run(const std::vector<cv::Mat>&  hiddens_,
+                    const std::vector<cv::Mat>&  features,
+                    const cv::gapi::GNetPackage& net,
+                    const size_t                 numClasses,
+                    const size_t                 endToken,
+                          std::vector<cv::Mat>&  res,
+                          cv::GComputation& graph) {
```

Comment:
Does it make sense to keep `GCompiled` in state?

Reply:
Parameter for cv::compile_args for main graph + tag for parameter --> cv::gapi::getCompileArg in setup.
Or, does ArrayDesc come in runtime?

Reply:
Is it for the net?

Reply:
Yes, probably..
Anyway - rewritten the kernel, it keeps `GCompiled` now. Added taking sizes from the net and passing it via `compile_args`